pkg/gardenlet/apis/config/validation/validation.go

```diff
@@ -85,6 +86,10 @@ func ValidateGardenletConfiguration(cfg *config.GardenletConfiguration, fldPath
 		for _, errorMessage := range validation.IsDNS1123Label(handler.Name) {
 			allErrs = append(allErrs, field.Invalid(handlerPath.Child("name"), handler.Name, errorMessage))
 		}
+
+		if !cfg.FeatureGates[string(features.APIServerSNI)] && handler.LoadBalancerService.LoadBalancerIP != nil {
+			allErrs = append(allErrs, field.Invalid(handlerPath.Child("loadBalancerService", "loadBalancerIP"), handler.LoadBalancerService.LoadBalancerIP, "cannot use loadbalacer ip when APIServerSNI feature gate is disabled"))
+		}
```

Comment:
You could also validate that specifying an empty string (`""`) is invalid, and potentially whether the specified IP address is valid?

Reply:
This is done. The empty string case is covered with `ip := net.ParseIp("")` where `ip` will be `nil` in case the passed string is empty or any other invalid ip format.